googletest/meson.build

```diff
@@ -3,6 +3,7 @@
 
 gtest = cpp.find_library('gtest', required: false)
 gtest_main = cpp.find_library('gtest_main', required: false)
+d3d12_lib = cpp.find_library('d3d12')
```

Comment:
These changes should be dropped now.

Reply:
Done

Reply:
From what I gathered when making the change I had, meson variables can work from the root down.  See this discussion, https://github.com/mesonbuild/meson/issues/2607 and this, https://stackoverflow.com/questions/55107396/meson-targets-that-depend-on-subdir-siblings .  I had defined ,`d3d12_lib = cpp.find_library('d3d12')`,  `dxcore_lib = cpp.find_library('dxcore', required: compiler_id == 'msvc')`, and `test_compile_opts` logic in the main build file because I wanted the test programs in the "test" and "googletest" to share those variables.

I originally had the stuff I described in the "googletest" and "test" subdir meson.build files but I deided to clean that up so that I don't have to check for "dxcore" twice.

Reply:
Right, but this code also does work on Linux, where not only is DXCore missing (from linker search dirs anyway) but so is D3D12. I missed that when reviewing your change.

I really need to get CI added to this repo at some point...